### ISSUE TITLE
SVG title not recognized as button text (incorrectly fails button-name check)

### ISSUE THREAD

**Issue by jordanlev** [dequelabs/axe-core#2650]
**Expectation**: button containing `<svg role="img"><title>...</title>...</svg>` satisfies button text requirement (`button-name` rule)

**Actual**: axe-core reports "critical: Buttons must have discernible text"

**Motivation**: An svg with role="img" and a title has wide screenreader support (for example, see "SVG pattern 5" section in https://www.deque.com/blog/creating-accessible-svgs/), and if this is the sole child of a button it should suffice as "button text". 

<pre><code>
axe-core version: 4.0.2
axe-webdriver, extension or other integration version: ?

Browser and Assistive Technology versions

For Tooling issues:
- Node version: 12.19.0
- Platform:  Mac
</code></pre>

**Comment by straker** [comment]
Sorry for the delay in getting back. I testing this is different browsers:

```html
<button>
  <svg role="img"><title>Accessible Name</title></svg>
</button>
```

- VO/Safari reads "button" (does not read the svg title)
- NVDA/Firefox reads "Button graphic accessible name"
- JAWS/IE11 reads "Accessible name button"
- JAWS/Chrome reads "Accessible name button"

As VO/Safari completely fails to read the title, I'm not sure we could accept just an svg and title as an accessible name.

**Comment by jordanlev** [comment]
Thanks for the thoughtful reply. I'm curious why your results seem to contradict what your Deque colleague Carrie Fisher found here: https://www.deque.com/blog/creating-accessible-svgs/ ("SVG Pattern 5"). Perhaps the svg needs to have other things on it (eg `<svg role="img" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">`), or perhaps it needs to contain a `<path>` element besides the title? (Or maybe that blog post needs to be corrected / clarified)?

**Comment by straker** [comment]
It's not so much the svg itself thats inaccessible, it's more about how the screen readers interpret the svg inside the button., Seems that VO/Safari doesn't compute the accessible name correctly.

**Comment by prios-jordan-lev** [comment]
I see... what I linked to wasn't wrapped in a button. Bummer. Well thank you again for the thoughtful replies.

**Comment by dylanb** [comment]
@straker Should we close this?

**Comment by straker** [comment]
@dylanb Ya, since this doesn't work for one of our accessibility supported browsers / screen readers, we'll still fail it as a technique.
